main.c

```diff
@@ -98,50 +161,28 @@ int main(int argc, char *argv[])
 				safe = true;
 			break;
 		case 'f':	/* next argument is program filename */
-			if (argv[1][2] != 0) {  /* arg is -fsomething */
-				if (npfile >= MAX_PFILE - 1)
-					FATAL("too many -f options");
-				pfile[npfile++] = &argv[1][2];
-			} else {		/* arg is -f something */
-				argc--; argv++;
-				if (argc <= 1)
-					FATAL("no program filename");
-				if (npfile >= MAX_PFILE - 1)
-					FATAL("too many -f options");
-				pfile[npfile++] = argv[1];
-			}
-			break;
+			fn = getarg(&argc, &argv, "no program filename");
+			if (npfile >= maxpfile) {
+				maxpfile += 20;
+				pfile = realloc(pfile,
+				    maxpfile * sizeof(*pfile));
```

Comment:
OK to join lines 167 and 168.

Reply:
Ok, I thought the speed limit was 80 :-)